### PR TITLE
[FIX] web: fix HTML user menu items on mobile client

### LINKED ISSUE #145693: markup Items not rendering correctly in mobile user menu
Impacted versions:
17.0+e

I have this code:

```js
import {UserMenu} from "@web/webclient/user_menu/user_menu";
import {patch} from "@web/core/utils/patch";
import {registry} from "@web/core/registry";
import {markup} from "@odoo/owl";

const userMenuRegistry = registry.category("user_menuitems");

function customItem(env) {
	return {
		type: "item",
		id: "custom_id",
		description: markup(
			`<div>
				My Custom Item
                <i class="fa fa-check px-2"></i>
            </div>`
		),
		callback: () => {
			console.log('Hi')
		},
		sequence: 1,
	};
}

patch(UserMenu.prototype, {
	setup() {
		super.setup();
		userMenuRegistry.add("custom-item", customItem, {force: true,})
	}
})

```
The code Is working as expected on desktop clients but on mobile devices, it shows the value instead of rendering it

Desktop:
Screen Shot 2023-12-09 at 5 16 08 PM

Mobile:
Screen Shot 2023-12-09 at 5 15 29 PM

### PR DESCRIPTION
**Pull request by Heshmatkhah**
HTML user menu items were escaped on the mobile client

Fixes #145693

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Heshmatkhah** [comment]
Hi @pparidans 
Can you check and see if it is okay or not?
Thank you

**Comment by aab-odoo** [comment]
@pparidans

**Comment by pparidans** [comment]
Hi @Heshmatkhah, first of all, sorry for the late reply...

Your fix looks good to me. :+1: 

May I ask you to add a test for it ? (and rebase of course :sweat_smile: )
Also, in the commit message, it would be good to mark that it wasn't useful to escape them on mobile and that on "desktop" it wasn't the case anyway.

**Comment by Heshmatkhah** [comment]
Hi @pparidans, thank you.
Thank you, rebase to which branch? 17.0, 18.0, master.
what should I do to fix this on other versions too?
I will add the test and change the commit message soon.

**Comment by pparidans** [comment]
> rebase to which branch? 17.0, 18.0, master. 

17.0 ; just to update your branch :wink: .
The forward-port will take care of the upper versions.

**Comment by Heshmatkhah** [comment]
Hi @pparidans
Done! check it please.
Thank you.

File: addons/web/static/tests/webclient/mobile/burger_user_menu_tests.js
Comment (pparidans) [inline addons/web/static/tests/webclient/mobile/burger_user_menu_tests.js:1]: Actually, there is already a QUnit test file for this component in `addons/web/static/tests/mobile/webclient/burger_menu/burger_menu_tests.js` :wink: 
Would you mind adding your test in that file ?
Thank you

**Review by pparidans — CHANGES_REQUESTED** [review]

File: addons/web/static/tests/webclient/mobile/burger_user_menu_tests.js
Comment (pparidans) [reply addons/web/static/tests/webclient/mobile/burger_user_menu_tests.js:1]: Hi, actually there are 2 JS unit-tests suites : one for the regular/desktop tests (cf. `/web/tests`) and one for the small-screen/mobile tests (cf. `/web/tests/mobile`).

Every file in `addons/web/static/tests/mobile/...` runs in the "mobile" test suite.

NB: I know, this is not really obvious... and is changed in Odoo 18+ with our new tests' runner... :wink:

Comment (Heshmatkhah) [reply addons/web/static/tests/webclient/mobile/burger_user_menu_tests.js:1]: Hi, @pparidans
Done!
Thanks for your advices.

**Review by pparidans — APPROVED** [review]
LGTM

Thank you for your fix :+1: 

@robodoo r+

**Pull request closed without merging**